### PR TITLE
DOC-5463: Improve description of INFER schema output

### PR DESCRIPTION
**Pull request by simon-dew**
Documentation issue: [DOC-5463](https://issues.couchbase.com/browse/DOC-5463)

The following documentation is ready for review: [INFER](https://github.com/couchbase/docs-server/blob/9f9139668bd48a3a4e696fa675494d54800fdc1a/modules/n1ql/pages/n1ql-language-reference/infer.adoc)

* Added syntax diagrams
* Updated schema output with `items`, `properties`, `$schema`, and `Flavor`
* Moved schema output after syntax

### CODE REVIEW

File: modules/n1ql/pages/n1ql-language-reference/infer.adoc [hunk 1]
```diff
@@ -73,26 +75,60 @@ For example, one of the Couchbase training modules has music tracks with a sub-d
...
+Each property is described by a key-value pair, in which the key is the name of the property, and the value gives recursive details of that property.
+
+Details for Documents and Subdocuments::
+[.out]`$schema`;; Specifies the version of the JSON Schema standard.
+[.out]`Flavor`;; Specifies the flavor of a document or subdocument.
+
+== RBAC Privileges
 
-User executing the INFER statement must have the _Query Select_ privilege granted on the target keyspace/bucket.
+The user executing the INFER statement must have the _Query Select_ privilege granted on the target keyspace/bucket.
```
Comment (amarantha-k) [inline modules/n1ql/pages/n1ql-language-reference/infer.adoc]: Use second person, something like
To execute the INFER statement, you must have....

File: modules/n1ql/pages/n1ql-language-reference/infer.adoc [hunk 2]
```diff
@@ -73,26 +75,60 @@ For example, one of the Couchbase training modules has music tracks with a sub-d
...
 For more details about user roles, see
 xref:learn:security/authorization-overview.adoc[Authorization].
 
 For example,
 
-To execute the following statement, user must have the _Query Select_ privilege on `pass:c[`travel-sample`]`.
+To execute the following statement, the user must have the _Query Select_ privilege on `pass:c[`travel-sample`]`.
```
Comment (amarantha-k) [inline modules/n1ql/pages/n1ql-language-reference/infer.adoc]: Similar to the previous comment, please use second person.

**Review by amarantha-k — APPROVED** [review]

File: modules/n1ql/pages/n1ql-language-reference/infer.adoc (hunk 1 above)
Comment (simon-dew) [reply modules/n1ql/pages/n1ql-language-reference/infer.adoc]: Done

File: modules/n1ql/pages/n1ql-language-reference/infer.adoc (hunk 2 above)
Comment (simon-dew) [reply modules/n1ql/pages/n1ql-language-reference/infer.adoc]: Done

**Pull request merged**